modules/core/src/main/java/org/apache/synapse/config/xml/EnrichMediatorFactory.java

```diff
@@ -187,4 +200,49 @@ public QName getTagQName() {
         return XML_Q;
     }
 
+    private void validateTypeCombination(OMElement sourceEle, OMElement targetEle) {
+        int sourceType = -1;
+        int targetType = -1;
+
+        // source type attribute
+        OMAttribute sourceTypeAttr = sourceEle.getAttribute(ATT_TYPE);
+        if (sourceTypeAttr != null && sourceTypeAttr.getAttributeValue() != null) {
```

Comment:
can sourceTypeAttr being null is a valid use case ? if not we need to throw an error.

Reply:
Done